site/docs/plugins/commands.md

```diff
@@ -5,10 +5,621 @@ next: false
+
+Therefore calling `setCommands` or `setMyCommands` with anything but lower_c4s3_commands will throw an exception.
+Commands not following this rules can still be registered, used and handled, but will never be displayed on the user menu as such.
+:::
+
+**Be aware** that `setCommands` and `setMyCommands` only affects the commands displayed in the user's commands menu, and not the actual access to them.
+You will learn how to implement restricted command access in the [Scoped Commands](#scoped-commands) section.
+
+### Grouping Commands
+
+Since we can split and group our commands into different instances, it allows for a much more idiomatic command file organization.
+
+Let's say we want to have developer-only commands.
+We can achieve that with the following code structure:
+
+```ascii
+src/
+├─ commands/
+│  ├─ admin.ts
+│  ├─ users/
+│  │  ├─ group.ts
+│  │  ├─ say-hi.ts
+│  │  ├─ say-bye.ts
+│  │  ├─ ...
+├─ bot.ts
+├─ types.d.ts
+tsconfig.json
+```
+
+::: tip Type Resolution
+For the sake of brevity, we are assuming your `tsconfig` file is well-set to resolve the types from `types.d.ts` and have resolved every other necessary import.
+:::
+
+The following code group exemplifies how we could implement a developer only command group, and update the Telegram client Command menu accordingly.
+Make sure you take notice of the different patterns being use in the `admin.ts` and `group.ts` file-tabs.
+
+::: code-group
+
+```ts [bot.ts]
+import { devCommands } from "./commands/admin.ts";
+import { userCommands } from "./commands/users/group.ts";
+
+export const bot = new Bot<MyContext>("MyBotToken");
+
+bot.use(commands());
+
+bot.use(userCommands);
+bot.use(devCommands);
+```
+
+```ts [admin.ts]
+import { userCommands } from './users/group.ts'
+
+export const devCommands = new CommandGroup<MyContext>()
+
+devCommands.command('devlogin', 'Greetings', async (ctx, next) => {
+   if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+      await ctx.reply('Hi to me')
+      await ctx.setMyCommands(userCommands, devCommands)
+   } else next()
+})
+
+devCommands.command('usercount', 'Greetings', async (ctx, next) => {
+   if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+      await ctx.reply(
+        `Active users: ${/** Your business logic */}`
+    )
+   } else next()
+})
+
+devCommands.command('devlogout', 'Greetings', async (ctx, next) => {
+    if (ctx.from?.id === ctx.env.DEVELOPER_ID) {
+       await ctx.reply('Bye to me')
+       await ctx.setMyCommands(userCommands)
+    } else next()
+ })
+```
+
+```ts [group.ts]
+import sayHi from "./say-hi.ts";
+import sayBye from "./say-bye.ts";
+import etc from "./another-command.ts";
+
+export const userCommands = new CommandGroup<MyContext>()
+  .add([sayHi, sayBye]);
+```
+
+```ts [say-hi.ts]
+export default new Command<MyContext>("sayhi", "Greetings", async (ctx) => {
+  await ctx.reply("Hello little User!");
+});
+```
+
+```ts [types.d.ts]
+type MyContext = Context & CommandsFlavor<MyContext>;
+```
```

Comment:
Can we simply delete this file and inline its type definition where needed?

Reply:
You're right, and I've just realized what you were saying. Yeah, we can remove it.

Reply:
Epic